src/screens/Assistant_v2/Results.tsx

```diff
@@ -112,7 +108,7 @@ const Results: React.FC<Props> = ({
   return (
     <View style={styles.container}>
       {tripPatterns?.map((tripPattern, i) => (
-        <Fragment key={String(tripPattern.id + i ?? i)}>
+        <Fragment key={i}>
```

Comment:
Jeg har lest meg opp litt mer på dette, og tror det er lurt å finne en unik identifikator i stedet for kun index. Si at man har fem reiseforslag i et søkeresultat, også blir søkeresultatet oppdatert 5 minutter etterpå og da er det første reiseforslaget forsvunnet. Jeg tror det er fare for at det i praksis blir det siste reiseforslaget som forsvinner fra visningen og ikke det første, siden basert på react key er det innslaget med key 4 som mangler. Vet ikke helt om det ble forståelig eller er helt presist, men sikkert kjekt å være føre var.

Reply:
stemmer det da? at reiseforslag forsvinner?

Reply:
Ja, de kan i utgangspunktet forsvinne. Keys burde aldri være index her, for det blir brukt som identifikatorer i mellom tilstandsendringer i React (eller i "rekonsillieringsfasen" av rendringen). Så for å finne endringer mellom tilstand A til tilstand B og hvilke endringer som må gjøres på det underliggende UI-laget (enten det er DOM-operasjoner eller programatiske operasjoner i RN her da). Så dersom du har ett element på posisjon 1 i tilstand A og gjør en endring som bytter ut det elementet med tilsvarende posisjon 1 i neste render, så vil det oppfattes som det samme elementet. Det kan få en del uheldige situasjoner. Det finnes noen _veldig_ få scenario hvor index kan brukes som keys, men de er sjelden nok til at tommelfingerregel er å ikke bruke index.

I tilfellet her så kan jeg se for meg flere bugs som kan skje som en konsekvens av det, særlig med tanke på at ting her kan oppdatere seg etterhvert som tiden går, du flytter din posisjon eller du gjør et nytt søk.

Reply:
enig!
Hadde en samtale med @gorandalum om dette og vi ble enige om å lage en unik ved å kombinere compressedQuery og PointsOnLink